src/tests/ftest/recovery/container_list_consolidation.py

```diff
@@ -0,0 +1,173 @@
+        actual_uuid = match.group(1)
+        expected_uuid = container.uuid.lower()
+        if actual_uuid != expected_uuid:
+            msg = "Unexpected container UUID! Expected = {}; Actual = {}".format(
+                expected_uuid, actual_uuid)
+            errors.append(msg)
+
+        # 6. Enable the checker.
+        self.log_step("Enable the checker.")
+        dmg_command.check_enable(stop=False)
+
+        # 7. Set policy to --all-interactive.
+        self.log_step("Set policy to --all-interactive.")
+        dmg_command.check_set_policy(all_interactive=True)
+
+        # 8. Start the checker and query the checker until the fault is detected.
+        self.log_step("Start and query the checker until the fault is detected.")
+        seq_num = None
+        # Start checker.
+        dmg_command.check_start()
+        # Query the checker until expected number of inconsistencies are repaired.
+        for _ in range(8):
+            check_query_out = dmg_command.check_query()
+            # Status is INIT before starting the checker.
+            if check_query_out["response"]["status"] == "RUNNING" and\
+                    check_query_out["response"]["reports"]:
+                seq_num = check_query_out["response"]["reports"][0]["seq"]
+                break
+            time.sleep(5)
+        if not seq_num:
+            self.fail("Checker didn't detect any fault!")
+
+        # 9. Repair by selecting the destroy option, 0.
+        msg = ("Repair with option 0; Destroy the orphan container to release space "
+               "[suggested].")
+        self.log_step(msg)
+        dmg_command.check_repair(seq_num=seq_num, action=0)
+
+        # 10. Query the checker until the fault is repaired.
+        self.log_step("Query the checker until the fault is repaired.")
+        repair_report = self.wait_for_check_complete()[0]
+
+        # Verify that the repair report has expected message "Discard the container".
+        action_message = repair_report["act_msgs"][0]
+        exp_msg = "Discard the container"
+        errors = []
+        if exp_msg not in action_message:
+            errors.append(f"{exp_msg} not in {action_message}!")
+
+        # 11. Disable the checker.
+        self.log_step("Disable the checker.")
+        dmg_command.check_disable(start=False)
+
+        # 12. Run the ddb command and verify that the container is removed from shard.
+        self.log_step(
+            "Run the ddb command and verify that the container is removed from shard.")
+        cmd_result = ddb_command.list_component()
+        ls_out = "\n".join(cmd_result[0]["stdout"])
+        uuid_regex = r"([0-f]{8}-[0-f]{4}-[0-f]{4}-[0-f]{4}-[0-f]{12})"
+        match = re.search(uuid_regex, ls_out)
+        if match:
+            errors.append("Container UUID is found in shard! Checker didn't remove it.")
+
+        # Start server to prepare for the cleanup.
+        dmg_command.system_start()
+
+        report_errors(test=self, errors=errors)
```

Comment:
Just pointing out that if system_start raises an exception and causes the test to fail, then we would miss the report_errors.

Reply:
Added `except CommandFailure` to catch the system start exception.